tests/integration-tests/syncTest.js

```diff
@@ -0,0 +1,257 @@
+import { Octokit } from "@octokit/rest";
+
+const githubToken = process.env.GITHUB_TOKEN;
+const apiKey = process.env.API_KEY;
+const prId = process.env.PR_ID;
+const targetRepoOwner = "Azure";
+const targetRepo = "azure-webpubsub"
+const octokit = new Octokit({
+    auth: githubToken,
+});
+
+function handleResponse(res) {
+    const fileNameRegex = /\*\*File Name:\*\*\s*`([^`]+)`/;
+    const fileNameMatch = res.match(fileNameRegex);
+    const fileName = fileNameMatch ? fileNameMatch[1] : "No file name found";
+
+    const fileContentRegex = /\*\*File Content:\*\*\s*\n*```([^`]+)```/;
+    const fileContentMatch = res.match(fileContentRegex);
+    const fileContent = fileContentMatch ? fileContentMatch[1].split("\n").slice(2).join("\n") : "No file content found";
+    return { fileName: `${fileName}`, fileContent: fileContent };
+}
+
+async function getLatestCommitSha(owner, repo) {
+    try {
+        const { data } = await octokit.rest.git.getRef({
+            owner,
+            repo,
+            ref: `heads/main`,
+        });
+        return data.object.sha;
+    } catch (error) {
+        console.error("Failed to get latest commit SHA:", error.message);
+        throw error;
+    }
+}
+
+async function createChangeBranch(owner, repo, sha) {
+    try {
+        const { data } = await octokit.rest.git.getRef({
+            owner,
+            repo,
+            ref: "refs/heads/[auto-generated]python integration test",
+        });
+        console.log("Branch already exists, using existing branch SHA:", data.object.sha);
+        return data.object.sha;
+    } catch (error) {
+        // If the branch does not exist, create new branch called "[auto-generated]python integration test"
+        if (error.status === 404) {
+            try {
+                const { data: newData } = await octokit.rest.git.createRef({
+                    owner,
+                    repo,
+                    ref: "refs/heads/[auto-generated]python integration test",
```

Comment:
branch name should have some unique id for the particular change

Reply:
branch name contains `[auto-generated]`? sounds strange

Reply:
how about `auto-generated-${branch name of original branch name}`?